packages/itwin/tree-widget/src/components/tree-header/TreeHeader.tsx

```diff
@@ -6,46 +6,66 @@
 import "./TreeHeader.scss";
 import classnames from "classnames";
 import { Children, useState } from "react";
-import { SvgMore } from "@itwin/itwinui-icons-react";
-import { ButtonGroup, DropdownMenu, IconButton, MenuItem } from "@itwin/itwinui-react";
-import { SearchBox } from "./SearchBox";
+import { SvgCaretDownSmall, SvgCaretUpSmall, SvgMore } from "@itwin/itwinui-icons-react";
+import { ButtonGroup, Divider, DropdownMenu, IconButton, MenuItem, SearchBox } from "@itwin/itwinui-react";
+import { TreeWidget } from "../../TreeWidget";
 
 import type { Viewport } from "@itwin/core-frontend";
-import type { SearchBoxProps } from "./SearchBox";
+import type { CommonProps } from "@itwin/core-react";
 
 /** @internal */
 export interface TreeHeaderButtonProps {
   viewport: Viewport;
 }
 
 /** @internal */
-export interface TreeHeaderProps extends Omit<SearchBoxProps,
-| "onIconClick"
-| "valueChangedDelay"
-| "searchOpen"
-| "onSearchOpen"
-| "onSearchClose"
-> {
+export interface TreeHeaderProps extends CommonProps {
+  /** Filtering is cleared after everything's loaded */
+  onFilterStart: (newFilter: string) => void;
+  /** listens for onClick event for Clear (x) icon */
+  onFilterClear: () => void;
+  /** Total number of results/entries */
+  resultCount?: number;
+  /** Current selected result index */
+  selectedIndex?: number;
+  /** Callback to currently selected result/entry change */
+  onSelectedChanged: (index: number) => void;
   /** Header buttons */
   children?: React.ReactNode;
 }
 
 /** @internal */
 export function TreeHeader(props: TreeHeaderProps) {
-  const { children, ...restProps } = props;
-  const [searchOpen, setSearchOpen] = useState(false);
-
+  const { onFilterStart, onFilterClear, resultCount, selectedIndex, onSelectedChanged, children, className } = props;
+  const [isSearchOpen, setIsSearchOpen] = useState<boolean>(false);
   return (
-    <div className={classnames("tree-widget-search-bar", props.className)}>
-      <HeaderButtons contracted={searchOpen}>{children}</HeaderButtons>
-      <div className="search-bar-search-container">
-        <SearchBox
-          {...restProps}
-          searchOpen={searchOpen}
-          onSearchOpen={() => setSearchOpen(true)}
-          onSearchClose={() => setSearchOpen(false)}
-        />
-      </div>
+    <div className={classnames("tree-widget-tree-header", className)}>
+      <HeaderButtons contracted={isSearchOpen}>
+        {children}
+      </HeaderButtons>
+      <SearchBox expandable onExpand={() => setIsSearchOpen(true)} onCollapse={() => setIsSearchOpen(false)} className={classnames("tree-widget-search-bar", !isSearchOpen && "contracted")}>
```

Comment:
I would suggest moving props into separate lines if there are more then 1 prop. It should make code easier to read.

Reply:
Maybe we can enforce [this rule](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-max-props-per-line.md) in the tree widget repo?

Reply:
fixed

Reply:
Yes, we can enable this rule